lib/src/value/calculation.dart

```diff
@@ -158,6 +344,201 @@ class SassCalculation extends Value {
+  /// arguments is an unquoted `var()` string.
+  static Value mod(Object dividend, Object? modulus) {
+    dividend = _simplify(dividend);
+    modulus = modulus.andThen(_simplify);
+    var args = [dividend, if (modulus != null) modulus];
+    _verifyLength(args, 2);
+    if (dividend is! SassNumber ||
+        modulus is! SassNumber ||
+        _arePossiblyCompatible(dividend, modulus)) {
+      return SassCalculation._("mod", args);
+    }
+    _verifyCompatibleNumbers(args);
+    return dividend.modulo(modulus);
+  }
+
+  /// Creates a `round()` calculation with the given [strategyOrNumber], [numberOrStep], and [step].
+  /// Strategy must be either nearest, up, down or to-zero.
+  ///
+  /// Number and step must be either a [SassNumber], a [SassCalculation], an
+  /// unquoted [SassString], a [CalculationOperation], or a
+  /// [CalculationInterpolation].
+  ///
+  /// This automatically simplifies the calculation, so it may return a
+  /// [SassNumber] rather than a [SassCalculation]. It throws an exception if it
+  /// can determine that the calculation will definitely produce invalid CSS.
+  ///
+  /// This may be passed fewer than two arguments, but only if one of the
+  /// arguments is an unquoted `var()` string.
+  static Value round(Object strategyOrNumber,
+      [Object? numberOrStep, Object? step]) {
+    var strategyOrNumberSimplified = _simplify(strategyOrNumber);
+    numberOrStep = numberOrStep.andThen(_simplify);
+    step = step.andThen(_simplify);
+
+    switch ((strategyOrNumberSimplified, numberOrStep, step)) {
+      case (SassNumber number, null, null):
+        return _matchUnits(number.value.round().toDouble(), number);
+      case (
+          SassString(text: 'nearest' || 'up' || 'down' || 'to-zero'),
+          SassString(isVar: false) || SassNumber(),
+          null
+        ):
+        throw SassScriptException("If strategy is not null, step is required.");
+
+      case (SassNumber number, SassNumber step, null):
+        _verifyCompatibleNumbers([number, step]);
+        return _roundWithStep(SassString('nearest'), number, step);
+
+      case (SassString strategy, SassNumber number, SassNumber step)
+          when {'nearest', 'up', 'down', 'to-zero'}.contains(strategy.text):
+        _verifyCompatibleNumbers([number, step]);
+        return _roundWithStep(strategy, number, step);
+
+      case (SassString(), _, _):
+      case (CalculationOperation(), _, _):
+      case (SassNumber(), _, null):
+        return SassCalculation._("round", [
+          strategyOrNumberSimplified,
+          if (numberOrStep != null) numberOrStep,
+          if (step != null) step
+        ]);
+
+      case (_, _?, _?):
+        throw SassScriptException(
+            "$strategyOrNumber must be either nearest, up, down or to-zero.");
+    }
+
+    throw ArgumentError(
+        "Expected arguments to be Object strategyOrNumber, Object? numberOrStep, Object? step).");
```

Comment:
This error doesn't make sense. You're just listing the type signature of the function here, there's no way the arguments can be anything other than this.